ceph.yaml

```diff
@@ -0,0 +1,342 @@
+        -DWITH_PYTHON3=${{vars.py-version}} \
+        -DENABLE_GIT_VERSION=ON \
+        -DLUA_LIBRARIES=/usr/lib/lua5.4/liblua.so \
+        -DWITH_BABELTRACE=OFF \
+        -DWITH_JAEGER=OFF \
+        -DWITH_LTTNG=OFF \
+        -DWITH_MANPAGE=ON \
+        -DWITH_RADOSGW_AMQP_ENDPOINT=OFF \
+        -DWITH_RADOSGW_SELECT_PARQUET=ON \
+        -DWITH_RADOSGW_POSIX=OFF \
+        -DWITH_RDMA=OFF \
+        -DWITH_REENTRANT_STRSIGNAL=ON \
+        -DWITH_SEASTAR=OFF \
+        -DWITH_SPDK=OFF \
+        -DWITH_SYSTEMD=ON \
+        -DWITH_CEPHFS_SHELL=ON \
+        -DWITH_SYSTEM_ARROW=ON \
+        -DWITH_SYSTEM_BOOST=OFF \
+        -DWITH_SYSTEM_FMT=OFF \
+        -DWITH_SYSTEM_LIBURING=ON \
+        -DWITH_SYSTEM_NPM=ON \
+        -DWITH_MGR_DASHBOARD_FRONTEND=ON \
+        -DWITH_SYSTEM_ROCKSDB=OFF \
+        -DWITH_SYSTEM_ZSTD=ON \
+        -DWITH_QATLIB=OFF \
+        -DWITH_QATZIP=OFF \
+        -DWITH_TESTS=OFF \
+        -DWITH_THREAD_SAFE_RES_QUERY=ON
+
+  - runs: cmake --build output --target common-options-objs
+
+  - uses: cmake/build
+
+  - uses: cmake/install
+
+  - uses: strip
+
+subpackages:
+  - name: ${{package.name}}-dev
+    dependencies:
+      runtime:
+        - ${{package.name}}
+    pipeline:
+      - uses: split/dev
+    test:
+      pipeline:
+        - uses: test/tw/ldd-check
+
+  - name: ${{package.name}}-libs
+    description: "libraries for ${{package.name}}"
+    dependencies:
+      provides:
+        - so:libceph-common.so.2=2
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.subpkgdir}}"/usr/lib/ceph
+          mv "${{targets.destdir}}"/usr/lib/lib*.so.* "${{targets.subpkgdir}}"/usr/lib
+          mv "${{targets.destdir}}"/usr/lib/ceph/lib*.so.* "${{targets.subpkgdir}}"/usr/lib/ceph
+    test:
+      pipeline:
+        - uses: test/tw/ldd-check
+
+  - name: ${{package.name}}-doc
+    pipeline:
+      - uses: split/manpages
+    description: ceph manpages
+    test:
+      pipeline:
+        - uses: test/docs
+
+test:
+  environment:
+    contents:
+      packages:
+        - uuidgen
+  pipeline:
+    - name: Ensure version build
+      runs: |
+        ceph --version  | grep ${{package.version}}
+        cephadm version | grep ${{package.version}}
+        ceph-fuse --version | grep ${{package.version}}
+        ceph-mds --version  | grep ${{package.version}}
+        ceph-mgr --version | grep ${{package.version}}
+        ceph-mon --version | grep ${{package.version}}
+        ceph-osd --version | grep ${{package.version}}
+        ceph-objectstore-tool --version | grep ${{package.version}}
+        radosgw --version | grep ${{package.version}}
+        radosgw-admin --version | grep ${{package.version}}
+        rbd --version | grep ${{package.version}}
+    - name: Bootstrap and validate Ceph cluster
+      runs: |
+        UUID=$(uuidgen)
+        HOSTNAME=$(hostname)
+
+        mkdir -p /etc/ceph
+        mkdir -p /var/run/ceph
+        mkdir -p /var/log/ceph
```

Comment:
We can either create these at the package level or using a post install scriptlet

Reply:
Moved into main package